src/screens/OtherProfile/hooks/useOtherProfileScreenHooks.tsx

```diff
@@ -0,0 +1,158 @@
+
+    try {
+      const otherProfileDataCache =
+        StorageUtils.otherProfileData.getForKey(targetUserProfileId) || '{}';
+      setOtherProfileData(JSON.parse(otherProfileDataCache));
+      if (otherProfileDataCache !== '{}') {
+        setIsProfileFetching(false);
+      }
+    } catch (e) {
+      console.log('error parsing JSON other profile data', e);
+    }
+
+    try {
+      const feedsCache = StorageUtils.otherProfileFeed.getForKey(targetUserProfileId) || '{}';
+
+      setOtherProfileFeed(JSON.parse(feedsCache), dispatchOtherProfile);
+    } catch (e) {
+      console.log('error parsing JSON other profile feed', e);
+    }
+
+    setIsBlocked(isBlockedFromCache);
+    setIsBlocking(isBlockingFromCache);
+  };
+
+  const fetchFeeds = async (offset = 0) => {
+    setIsLoading(true);
+    try {
+      const result = await getOtherFeedsInProfile(targetUserProfileId, offset);
+      const {data: feedOtherProfile} = result;
+      const mapNewData = mappingColorFeed(feedOtherProfile);
+      if (offset === 0) {
+        setOtherProfileFeed(mapNewData, dispatchOtherProfile);
+        StorageUtils.otherProfileFeed.setForKey(targetUserProfileId, JSON.stringify(mapNewData));
+      } else {
+        const clonedFeeds = [...feeds, ...mapNewData];
+        setOtherProfileFeed(clonedFeeds, dispatchOtherProfile);
+        StorageUtils.otherProfileFeed.setForKey(targetUserProfileId, JSON.stringify(clonedFeeds));
+      }
+      setPostOffset(Number(result.offset));
+    } catch (e) {
+      console.log('error', e);
+    }
+    setIsLoading(false);
+  };
+
+  const fetchOtherProfileData = async () => {
+    try {
+      const result = await getOtherProfile(username);
+      if (result.code === 200) {
+        const data = result?.data;
+        setOtherProfileData(data);
+        setOtherProfileUserId(data.user_id);
+
+        StorageUtils.otherProfileData.setForKey(targetUserProfileId, JSON.stringify(data));
+      }
+    } catch (e) {
+      console.log('error fetching other profile data', e);
+      if (e?.response?.data?.message) {
+        SimpleToast.show(e.response.data.message, SimpleToast.SHORT);
+      }
+    }
+  };
+
+  const fetchBlockStatus = async (withFeedsFetching = false) => {
+    const sendData = {
+      user_id: targetUserProfileId
+    };
+    try {
+      const processGetBlock = await checkUserBlock(sendData);
+      if (processGetBlock.status === 200) {
+        const isBlockedFromApi = processGetBlock.data.data.blocked;
+        const isBlockingFromApi = processGetBlock.data.data.blocker;
+
+        setIsBlocked(isBlockedFromApi);
+        setIsBlocking(isBlockingFromApi);
+
+        StorageUtils.blockedStatus.setForKey(targetUserProfileId, isBlockedFromApi.toString());
+        StorageUtils.blockingStatus.setForKey(targetUserProfileId, isBlockingFromApi.toString());
+
+        if (withFeedsFetching) await fetchFeeds();
+      }
+    } catch (e) {
+      console.log('error fetching block status', e);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous point, these catch blocks only log the error. Consider adding more robust error handling here.



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
closed